sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5373,6 +5372,27 @@ pi_result piextUSMSharedAlloc(void **ResultPtr, pi_context Context,
   return PI_SUCCESS;
 }
 
+pi_result piextUSMHostAlloc(void **ResultPtr, pi_context Context,
+                            pi_usm_mem_properties *Properties, size_t Size,
+                            pi_uint32 Alignment) {
+  if (!UseUSMAllocator ||
+      // L0 spec says that allocation fails if Alignment != 2^n, in order to
+      // keep the same behavior for the allocator, just call L0 API directly and
+      // return the error code.
+      ((Alignment & (Alignment - 1)) != 0)) {
+    return USMHostAllocImpl(ResultPtr, Context, Properties, Size, Alignment);
+  }
+
+  try {
+    *ResultPtr = Context->HostMemAllocContext->allocate(Size, Alignment);
+  } catch (const UsmAllocationException &Ex) {
+    *ResultPtr = nullptr;
+    return Ex.getError();
+  }
```

Comment:
Let's catch other exceptions too (and die), so that nothing is thrown from a PI call

Reply:
OK